pkgs/applications/version-management/sapling/default.nix

```diff
@@ -123,6 +125,25 @@ let
       sed -i "s|https://files.pythonhosted.org/packages/[[:alnum:]]*/[[:alnum:]]*/[[:alnum:]]*/|file://$NIX_BUILD_TOP/$sourceRoot/hack_pydeps/|g" $sourceRoot/setup.py
     '';
 
+    # Now, copy the "sl web" (aka edenscm-isl) results into the output of this
+    # package, so that the command can actually work. NOTES:
+    #
+    # 1) This applies on all systems (so no conditional a la postFixup)
+    # 2) This doesn't require any kind of fixup itself, so we leave it out
+    #    of postFixup for that reason, too
+    # 3) If asked, we optionally patch in a hardcoded path to the 'nodejs' package,
+    #    so that 'sl web' always works
+    # 4) 'sl web' will still work if 'nodejs' is in $PATH, just not OOTB
+    preFixup = ''
+      sitepackages=$out/lib/${python38Packages.python.libPrefix}/site-packages
```

Comment:
It may be compatible with 3.11 now, seeing https://github.com/facebook/sapling/commit/81685da429b7245f939792506d2b3568db1a19fe. When I update the package version I can make these changes.

Reply:
3.10 would be fine as well and even bumping it to 3.9 would improve it a little.

Reply:
I found this very interesting and informative guide: https://devguide.python.org/versions/

I'm guessing that Nixpkgs that's "bugfix+" level, not security+. Either way, I don't see a reason to linger on older versions; we should be able to fix this in pretty short order I think, but I'm pretty busy this week.